interop-cats/jvm/src/main/scala/scalaz/zio/interop/stm/STM.scala

```diff
@@ -0,0 +1,223 @@
+
+  /**
+   * See [[scalaz.zio.stm.STM#filter]]
+   */
+  final def filter(f: A => Boolean): STM[F, A] =
+    collect {
+      case a if f(a) => a
+    }
+
+  /**
+   * See [[scalaz.zio.stm.STM#flatMap]]
+   */
+  final def flatMap[B](f: A => STM[F, B]): STM[F, B] = new STM(underlying.flatMap(f.andThen(_.underlying)))
+
+  /**
+   * See [[scalaz.zio.stm.STM#flatten]]
+   */
+  final def flatten[B](implicit ev: A <:< STM[F, B]): STM[F, B] =
+    self flatMap ev
+
+  /**
+   * See [[scalaz.zio.stm.STM#fold]]
+   */
+  final def fold[B](f: Throwable => B, g: A => B): STM[F, B] = new STM(underlying.fold(f, g))
+
+  /**
+   * See [[scalaz.zio.stm.STM#foldM]]
+   */
+  final def foldM[B](f: Throwable => STM[F, B], g: A => STM[F, B]): STM[F, B] =
+    new STM(underlying.foldM(f.andThen(_.underlying), g.andThen(_.underlying)))
+
+  /**
+   * See [[scalaz.zio.stm.STM#map]]
+   */
+  final def map[B](f: A => B): STM[F, B] = new STM(underlying.map(f))
+
+  /**
+   * See [[scalaz.zio.stm.STM#mapError]]
+   */
+  final def mapError[E1 <: Throwable](f: Throwable => E1): STM[F, A] = new STM(underlying.mapError(f))
+
+  /**
+   * See [[scalaz.zio.stm.STM#option]]
+   */
+  final def option: STM[F, Option[A]] =
+    fold[Option[A]](_ => None, Some(_))
+
+  /**
+   * See [[scalaz.zio.stm.STM#orElse]]
+   */
+  final def orElse[A1 >: A](that: => STM[F, A1]): STM[F, A1] = new STM(underlying.orElse(that.underlying))
+
+  /**
+   * See [[scalaz.zio.stm.STM#orElseEither]]
+   */
+  final def orElseEither[B](that: => STM[F, B]): STM[F, Either[A, B]] =
+    (self map (Left[A, B](_))) orElse (that map (Right[A, B](_)))
+
+  /**
+   * See scalaz.zio.stm.STM.unit
+   */
+  final def unit: STM[F, Unit] = const(())
+
+  /**
+   * See scalaz.zio.stm.STM.unit
+   */
+  final def void: STM[F, Unit] = unit
+
+  /**
+   * Same as [[filter]]
+   */
+  final def withFilter(f: A => Boolean): STM[F, A] = filter(f)
+
+  /**
+   * See [[scalaz.zio.stm.STM#zip]]
+   */
+  final def zip[B](that: => STM[F, B]): STM[F, (A, B)] =
+    (self zipWith that)((a, b) => a -> b)
+
+  /**
+   * See [[scalaz.zio.stm.STM#zipLeft]]
+   */
+  final def zipLeft[B](that: => STM[F, B]): STM[F, A] =
+    (self zip that) map (_._1)
+
+  /**
+   * See [[scalaz.zio.stm.STM#zipRight]]
+   */
+  final def zipRight[B](that: => STM[F, B]): STM[F, B] =
+    (self zip that) map (_._2)
+
+  /**
+   * See [[scalaz.zio.stm.STM#zipWith]]
+   */
+  final def zipWith[B, C](that: => STM[F, B])(f: (A, B) => C): STM[F, C] =
+    self flatMap (a => that map (b => f(a, b)))
+
+  /**
+   * Switch from effect F to effect G using transformation `f`.
+   */
+  final def mapK[G[+ _]](f: F ~> G): STM[G, A] = new STM(underlying)(f compose liftIO)
+}
+
+object STM {
+
+  final def succeed[F[+ _], A](a: A)(implicit liftIO: ZIO[Any, Throwable, ?] ~> F): STM[F, A] = new STM(ZSTM.succeed(a))
+
+  final def atomically[F[+ _], A](stm: STM[F, A])(implicit liftIO: ZIO[Any, Throwable, ?] ~> F): F[A] =
```

Comment:
This can make do with an implicit `Runtime[Any]`, then no need for implicit `FunctionK`.